framework/src/Volo.Abp.Core/Volo/Abp/Internal/Telemetry/EnvironmentInspection/Core/SoftwareDetector.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Diagnostics;
+using System.Text;
+using System.Threading.Tasks;
+using Volo.Abp.DependencyInjection;
+using Volo.Abp.Internal.Telemetry.EnvironmentInspection.Contracts;
+
+namespace Volo.Abp.Internal.Telemetry.EnvironmentInspection.Core;
+
+[ExposeServices(typeof(ISoftwareDetector))]
+abstract internal class SoftwareDetector: ISoftwareDetector , ISingletonDependency
+{
+    public abstract string Name { get; }
+    public abstract Task<SoftwareInfo?> DetectAsync();
+
+    protected virtual async Task<string?> ExecuteCommandAsync(string command, string? arg)
+    {
+        var outputBuilder = new StringBuilder();
+
+        var processStartInfo = new ProcessStartInfo
```

Comment:
It would be better to use `CmdHelper` here. There is a bug/problem running commands in MAC computers, so fixing it in `CmdHelper` will prevent future problems about data collection.